libs/platform/src/lib/components/action-bar/action-bar.component.ts

```diff
@@ -1,135 +1,58 @@
-
-    /**
-     * Actionbar description
-     */
-    @Input()
-    description: string;
-
-    /**
-     * Show "back" button.
-     */
-    @Input()
-    showBackButton = false;
-
-    @Input()
-    showOnlyMenu = false;
-
-    /**
-     * "back" button label.
-     */
-    @Input()
-    backButtonLabel = 'Go Back';
-    /**
-     * used to specify the posistion of the menu1`
-     */
-    @Input()
-    placement: Placement;
-
-    /**
-     * List of action items.
-     */
-    @Input()
-    actionItems: ActionItem[];
-
-    /**
-     * View child of action bar component
-     */
-    @ViewChild('inputTitle') private inputTitle: ElementRef;
-
-    /**
-     * Emitted event when "back" button is clicked.
-     */
-    @Output()
-    backButtonClick: EventEmitter<void> = new EventEmitter();
-
-    /**
-     * Emitted event when input textbox out of focus.
-     */
-
-    @Output()
-    titleRenamed: EventEmitter<string> = new EventEmitter<string>();
-
-    /**
-     * Emitted event when action button is clicked.
-     */
-    @Output()
-    itemClick: EventEmitter<ActionItem> = new EventEmitter<ActionItem>();
-    timer;
-
-    constructor(private cd: ChangeDetectorRef) {}
-
-    ngOnInit() {}
-
-    enableEditTitle(editing: boolean) {
-        this.editing = editing;
-        this.timer = setTimeout(() => this.inputTitle.nativeElement.focus(), 0);
-        this.cd.markForCheck();
-    }
-
-    actionItemClicked(item: ActionItem) {
-        this.itemClick.emit(item);
-        this.cd.markForCheck();
-    }
-
-    onFocusOut() {
-        this.editing = false;
-        this.titleRenamed.emit(this.title);
-        this.cd.markForCheck();
-    }
-    outSideClick = ($event: Event) => {
-        if (!this.editing) {
-            this.editing = false;
-            this.cd.markForCheck();
-        }
-    };
-
-    ngOnDestroy() {
-        if (this.timer) {
-            clearTimeout(this.timer);
-        }
-    }
-}
+import {
+    Component,
+    ChangeDetectionStrategy,
+    Input,
+    Output,
+    EventEmitter,
+    ChangeDetectorRef
+} from '@angular/core';
+import { BaseComponent } from '../base';
+@Component({
+    selector: 'fdp-action-bar',
+    templateUrl: './action-bar.component.html',
+    styleUrls: ['./action-bar.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ActionBarComponent extends BaseComponent {
+    /**
+     * Actionbar title
+     */
+
+    @Input()
+    title: string;
+
+    /**
+     * Actionbar description
+     */
+    @Input()
+    description: string;
+
+
+    /**
+     * Compact mode
+     */
+    @Input()
+    compact = true;
+
+    /**
+     * Show "back" button.
+     */
+    @Input()
+    showBackButton = false;
+
+    /**
+     * "back" button label.
+     */
+    @Input()
+    backButtonLabel = 'Go Back';
+
+    /**
+     * Emitted event when "back" button is clicked.
+     */
+    @Output()
+    backButtonClick: EventEmitter<void> = new EventEmitter();

```

Comment:
who is using this output ?

Reply:
Application team is going to use , to perform action when back button click